src/lib/formats/dvk_mx_dsk.cpp

```diff
@@ -23,65 +23,45 @@
 #include "ioprocs.h"
 #include "multibyte.h"
 
+constexpr uint32_t ID_PATTERN = 0xaaaaffaf;
 
 const floppy_image_format_t::desc_e dvk_mx_format::dvk_mx_new_desc[] = {
 	/* 01 */ { FM, 0x00, 8*2 }, // eight 0x0000 words
-	/* 03 */ { FM, 0x00, 1 },
-	/* 02 */ { FM, 0xf3, 1 },   // word 0x00f3
-	/* 05 */ { FM, 0x00, 1 },
-	/* 04 */ { TRACK_ID_FM },   // track number word
-	/* 05 */ { SECTOR_LOOP_START, 0, 10 },  // 11 sectors
+	/* 02 */ { FM, 0x00, 1 },
+	/* 03 */ { FM, 0xf3, 1 },   // word 0x00f3
+	/* 04 */ { FM, 0x00, 1 },
+	/* 05 */ { TRACK_ID_FM },   // track number word
+	/* 06 */ { SECTOR_LOOP_START, 0, 10 },  // 11 sectors
 	/* 07 */ {   SECTOR_DATA_MX, -1 },
-	/* 10 */ { SECTOR_LOOP_END },
-	/* 13 */ { FM, 0x83, 1 },
+	/* 08 */ { SECTOR_LOOP_END },
+	/* 09 */ { FM, 0x83, 1 },
+	/* 10 */ { OFFSET_ID_FM },
+	/* 11 */ { FM, 0x83, 1 },
 	/* 12 */ { OFFSET_ID_FM },
-	/* 15 */ { FM, 0x83, 1 },
+	/* 13 */ { FM, 0x83, 1 },
 	/* 14 */ { OFFSET_ID_FM },
-	/* 17 */ { FM, 0x83, 1 },
-	/* 16 */ { OFFSET_ID_FM },
-	/* 18 */ { END }
+	/* 15 */ { END }
 };
 
 const floppy_image_format_t::desc_e dvk_mx_format::dvk_mx_old_desc[] = {
 	/* 01 */ { FM, 0x00, 30*2 },
-	/* 03 */ { FM, 0x00, 1 },
-	/* 02 */ { FM, 0xf3, 1 },   // word 0x00f3
-	/* 05 */ { FM, 0x00, 1 },
-	/* 04 */ { TRACK_ID_FM },   // track number word
+	/* 02 */ { FM, 0x00, 1 },
+	/* 03 */ { FM, 0xf3, 1 },   // word 0x00f3
+	/* 04 */ { FM, 0x00, 1 },
+	/* 05 */ { TRACK_ID_FM },   // track number word
 	/* 06 */ { SECTOR_LOOP_START, 0, 10 },  // 11 sectors
 	/* 07 */ {   SECTOR_DATA_MX, -1 },
-	/* 10 */ { SECTOR_LOOP_END },
-	/* 13 */ { FM, 0x83, 1 },
-	/* 11 */ { FM, 0x01, 1 },
-	/* 15 */ { FM, 0x83, 1 },
-	/* 14 */ { FM, 0x01, 1 },
-	/* 16 */ { END }
+	/* 08 */ { SECTOR_LOOP_END },
+	/* 09 */ { FM, 0x00, 1 },
+	/* 10 */ { FM, 0116, 1 },
+	/* 11 */ { FM, 0x00, 4*2 }, // four 0x0000 words
+	/* 12 */ { END }
```

Comment:
I'm convinced at this point that this pseudolanguage was a mistake and direction doing fm_w() and friends is better, but as long as it's there you're welcome to use it.